addon/components/-animated-orphans.js

```diff
@@ -0,0 +1,355 @@
+    if (!this._inserted) { return []; }
+    return Array.from(this.element.children).map(element => {
+      let child = this._elementToChild.get(element);
+      if (child.shouldRemove) {
+        // child was not animating in the previously interrupted
+        // animation, so its safe to remove
+        element.remove();
+      } else {
+        let sprite = Sprite.positionedStartingAt(element, ownSprite);
+        sprite.owner = child;
+        // we need to flag each existing child for removal at the
+        // start of each animation. That's what reinitializes its
+        // removal blockers count.
+        child.flagForRemoval();
+        return sprite;
+      }
+    }).filter(Boolean);
+  },
+
+  _groupActiveSprites(activeSprites) {
+    let groups = [];
+    for (let sprite of activeSprites) {
+      let { transition, duration } = this._childToTransition.get(sprite.owner);
+      let group = groups.find(g => g.transition === transition);
+      if (!group) {
+        group = { transition, duration, sprites: []};
+        groups.push(group);
+      }
+      group.sprites.push(sprite);
+    }
+    return groups;
+  },
+
+  _prepareSprite(sprite) {
+    sprite.hide();
+    let newElement = sprite.element.cloneNode(true);
+    continueMotions(sprite.element, newElement);
+    sprite.element = newElement;
+    return sprite;
+  },
+
+  _onFirstMotionStart(activeSprites, cycle, sprite) {
+    if (activeSprites.indexOf(sprite) === -1) {
+
+      // in most animators, sprites are still living in their normal place in
+      // the DOM, and so they will necessarily start out with their appearance
+      // matching initialComputedStyles. But here we're dealing with an orphan
+      // who may have lost inherited styles. So we manually re-apply the
+      // initialComputedStyles that were snapshotted before it was moved. See
+      // COPIED_CSS_PROPERTIES to see exactly which property we copy (we don't
+      // do all of them, that sounds expensive).
+      //
+      // Also, unfortunately getComputedStyles has legacy behavior for
+      // line-height that gives us the "used value" not the "computed value".
+      // The used value doesn't inherit correctly, so we can't set it here, so
+      // we're pulling that one out.
+      let s = Object.assign({}, sprite.initialComputedStyle);
+      delete s['line-height'];
+      sprite.applyStyles(s);
+
+      this.element.appendChild(sprite.element);
+      sprite.lock();
+      sprite.reveal();
+      activeSprites.push(sprite);
+      this._elementToChild.set(sprite.element, sprite.owner);
+    }
+    sprite.owner.block(cycle);
+  },
+
+  _onMotionStart(cycle, sprite) {
+    sprite.reveal();
+    sprite.owner.block(cycle);
+  },
+
+  _onMotionEnd(cycle, sprite) {
+    sprite.owner.unblock(cycle);
+  }
+
+});
+
+function * ancestorsOf(component) {
+  let pointer = component.parentView;
+  while (pointer) {
+    yield pointer;
+    pointer = pointer.parentView;
+  }
+}
```

Comment:
This is copied over from the motion service, might want to extract or export & reuse.